### ISSUE TITLE
Relation extraction in own KIE dataset

### ISSUE THREAD

**Issue by Tanmay98** [PaddlePaddle/PaddleOCR#8449]
Hi, 
After reading in the documentation, I tried annotating 45-50 images for finetuning the KIE model for own dataset, but the results for the SER model seems fine whereas RE was not working properly. 

Can you tell me what might be the issue, my guess is :

- either I need more training samples for KIE RE model
- or did I not train the RE model properly itself with 50 images?

**Comment by Prasanna-iq** [comment]
Did you added linking between the keys?. also can you share the dataset for more debugging?

**Comment by Tanmay98** [comment]
Yes I did add the linking parameter (ids in a array for which I want the relation to exist).  Below I have attaached my train.json file. Let me know if there is something wrong.

https://drive.google.com/file/d/11ReQRDPoh5Xdy-dMgYf0DL4bH4iNJ3Ek/view?usp=sharing

**Comment by Prasanna-iq** [comment]
Dataset looks fine fine to me. did you check the model Inference results with the training images?.

if it is giving RE in training data then the model is overfitted. else your doing something wrong with the training step

**Comment by Tanmay98** [comment]
Hi, I am attaching the train.log file for my ser_re model training and some sample inferneces.

Train.log: https://drive.google.com/file/d/1vLWopGaaR1rr46MQSb4uo4cy9RfGtQGg/view?usp=sharing

![temp2_ser_re](https://user-images.githubusercontent.com/31308303/204274510-14b3212a-fdad-4298-892c-d178bc5dcc7c.jpg)

**Comment by Tanmay98** [comment]
Also, @Prasanna-iq the inference results for the SER model were decent but the re model was performing poorly

**Comment by Prasanna-iq** [comment]
you did not trained the model properly. check your image path with the below error. i think you have missed a backslash.
  
  
  File "/Users/akhilbaweja/mca_ocr/PaddleOCR/ppocr/data/simple_dataset.py", line 132, in __getitem__
    raise Exception("{} does not exist!".format(img_path))
Exception: train_data/mca/mca_train/image/8907f4ae-dir_2_pdf6_page7.jpg does not exist!

**Comment by Tanmay98** [comment]
Hi @Prasanna-iq , I think it contains old logs too, so if you go at the bottom, the training started as the data was loaded correctly. Can you check once again?

**Comment by littletomatodonkey** [comment]
Hi, you should check the SER results firstly (with the gt ocr information), if they are all right, you can use SER+RE (with the gt ocr information again) to check the RE performance, as we can see in your log, it's high enough, so the performace might be good. Finally, you can use the OCR model to get the ocr information, and if the results are worse, then the OCR model training is your most important thing to do.

**Comment by Tanmay98** [comment]
Thanks @littletomatodonkey for your suggestion ! 
I will look into inferencing each of the models ocr, SER, and RE once again. and update

**Comment by TasneemVKhan** [comment]
@Tanmay98 How did you label your dataset ? Please can you help?

**Comment by piarosebelledelapaz** [comment]
@Tanmay98 hello, i am currently training SER and RE models using custom dataset. I have some few questions to ask you, would you be available to contact via email? Otherwise here are some of my questions:

1) did you change the classlist accordingly to your dataset? if so, i saw that you have still the QUESTION-ANSWER pairs, so i want to ask if you just used this as your class labels.
2) did you change anything specifically to the code to fit your dataset?
